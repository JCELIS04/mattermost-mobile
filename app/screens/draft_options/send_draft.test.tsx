// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {General, Screens} from '@constants';
import {dismissBottomSheet} from '@screens/navigation';
import {fireEvent, renderWithIntlAndTheme} from '@test/intl-test-helper';

import SendDraft from './send_draft';

jest.mock('@screens/navigation', () => ({
    dismissBottomSheet: jest.fn(),
}));

describe('Send Draft', () => {
    it('should render the component', () => {
        const props = {
            channelId: 'channel_id',
            channelName: 'channel_name',
            rootId: '',
            channelType: General.OPEN_CHANNEL,
            bottomSheetId: Screens.DRAFT_OPTIONS,
            currentUserId: 'current_user_id',
            maxMessageLength: 4000,
            useChannelMentions: true,
            userIsOutOfOffice: false,
            customEmojis: [],
            value: 'value',
            files: [],
            postPriority: '' as unknown as PostPriority,
            persistentNotificationInterval: 0,
            persistentNotificationMaxRecipients: 0,
            draftReceiverUserName: undefined,
        };
        const wrapper = renderWithIntlAndTheme(
            <SendDraft
                {...props}
            />,
        );
        const {getByText} = wrapper;
        expect(getByText('Send draft')).toBeTruthy();
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should call dismissBottmSheet after sending the draft', () => {
        const props = {
            channelId: 'channel_id',
            channelName: 'channel_name',
            rootId: '',
            channelType: General.OPEN_CHANNEL,
            bottomSheetId: Screens.DRAFT_OPTIONS,
            currentUserId: 'current_user_id',
            maxMessageLength: 4000,
            useChannelMentions: true,
            userIsOutOfOffice: false,
            customEmojis: [],
            value: 'value',
            files: [],
            postPriority: '' as unknown as PostPriority,
            persistentNotificationInterval: 0,
            persistentNotificationMaxRecipients: 0,
            draftReceiverUserName: undefined,
        };
        const wrapper = renderWithIntlAndTheme(
            <SendDraft
                {...props}
            />,
        );
        const {getByTestId} = wrapper;
        fireEvent.press(getByTestId('send_draft_button'));
        expect(dismissBottomSheet).toHaveBeenCalledTimes(1);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
