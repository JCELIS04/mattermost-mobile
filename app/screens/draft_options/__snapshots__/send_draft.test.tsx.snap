// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send Draft should call dismissBottmSheet after sending the draft 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 16,
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
  testID="send-draft-button"
>
  <Icon
    color="rgba(63,67,80,0.56)"
    name="send"
    size={24}
  />
  <Text
    style={
      {
        "color": "#3f4350",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      }
    }
  >
    Send draft
  </Text>
</View>
`;

exports[`Send Draft should call sendDraft when pressed 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 16,
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
  testID="send-draft-button"
>
  <Icon
    color="rgba(63,67,80,0.56)"
    name="send"
    size={24}
  />
  <Text
    style={
      {
        "color": "#3f4350",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      }
    }
  >
    Send draft
  </Text>
</View>
`;

exports[`Send Draft should render the component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 16,
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
  testID="send-draft-button"
>
  <Icon
    color="rgba(63,67,80,0.56)"
    name="send"
    size={24}
  />
  <Text
    style={
      {
        "color": "#3f4350",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      }
    }
  >
    Send draft
  </Text>
</View>
`;
